t/security.t

```diff
@@ -0,0 +1,61 @@
+use Mojo::Base -strict;
+use Test::Mojo;
+use Test::More;
+
+use Mojolicious::Lite;
+post '/protected' => sub {
+  my $c = shift->openapi->valid_input or return;
+  $c->reply->openapi(200 => {ok => 1});
+  },
+  'protected';
+
+plugin OpenAPI => {
+  url      => 'data://main/sec.json',
+  security => {
+    dummy => sub {
+      my ($c, $config, $next) = @_;
+      return $c->$next if $c->req->headers->authorization;
+      return $c->render(json => $config, status => 401);
```

Comment:
Yes.
My point was to catch the programmer error where one doesn't render anything by mistake or by not understanding the documentation, if the auth fails, and the $c->$next is not called.

I just wanted to make sure that the programmer gets some other error than "nothing rendered" if he doesn't understand that you must render something on a failed authentication. "nothing rendered" is really really vague and tracking the source of this problem takes ages.

This can prolly be dealt with in _security_action()  ?